src/ansible_compat/runtime.py

```diff
@@ -199,6 +203,31 @@ def install_collection(
             _logger.error(msg)
             raise InvalidPrerequisiteError(msg)
 
+    def install_collection_from_disk(
+        self, path: str, destination: Optional[Union[str, pathlib.Path]] = None
+    ) -> None:
+        """Build and install collection from a given disk path."""
```

Comment:
Also, since you have destination passed down to all invocations the same way, consider defining a partial like this:
```suggestion
        """Build and install collection from a given disk path."""
        install_collection = functools.partial(self.install_collection, destination=destination)
```

Reply:
I do not think is really needed, in would make the code harder to read, partials are not really the most popular python constructs.